### ISSUE TITLE
W1b: `bbox_to_mask3d` returns float32 `(B, 1, D, H, W)` while `bbox_to_mask` and `Boxes3D.to_mask` keep the input dtype with no channel axis

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4250]
The three mask builders already differ in argument order (#4014) and rounding policy (#4015). They also disagree on the output dtype and rank: `bbox_to_mask3d` ends with `.float()` and keeps a singleton channel axis from its intermediate broadcast, so it returns `float32` `(B, 1, D, H, W)` for any input dtype (float64, float16, bfloat16, int64 all executed), while `bbox_to_mask` returns `(B, H, W)` in the input dtype (int64 in gives int64 out) and `Boxes3D.to_mask` returns `(N, D, H, W)` in the box dtype.

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0, CPU and MPS.

## Reproduction

```python
import torch
from kornia.geometry.bbox import bbox_to_mask, bbox_to_mask3d
from kornia.geometry.boxes import Boxes3D

cuboid = torch.tensor([[[1., 1., 1.], [3., 1., 1.], [3., 3., 1.], [1., 3., 1.],
                        [1., 1., 2.], [3., 1., 2.], [3., 3., 2.], [1., 3., 2.]]], dtype=torch.float64)
square = torch.tensor([[[1., 1.], [3., 1.], [3., 2.], [1., 2.]]], dtype=torch.float64)
for label, out in (("bbox_to_mask3d", bbox_to_mask3d(cuboid, (4, 5, 6))),
                   ("Boxes3D.to_mask", Boxes3D(cuboid).to_mask(4, 5, 6)),
                   ("bbox_to_mask", bbox_to_mask(square, 6, 5))):
    print(label, tuple(out.shape), out.dtype)
```

## Observed

```text
bbox_to_mask3d (1, 1, 4, 5, 6) torch.float32
Boxes3D.to_mask (1, 4, 5, 6) torch.float64
bbox_to_mask (1, 5, 6) torch.float64
```

## Expected

One output contract across the family: the mask in the input dtype (idiom 2, a result that does not inherit its input's dtype) and one rank convention. Changing `bbox_to_mask3d`'s dtype or dropping its channel axis is a behavior change for its callers, so it belongs with #4015 in the coordinated Repair & Deprecation window; batch 4c documents the current output on both symbols and pins it (`test_wart_bbox_to_mask3d_returns_float32_with_a_channel_axis_4250`).

## Classification

Semantic contract / dtype and rank policy, not broken output. Idiom 2 (hard-coded `float32`), plus a family contract split.

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: the input dtype and the `Boxes3D.to_mask` rank (no channel axis), decided together with the shared implementation proposed in #4015. Shape: align contract, all names kept. Divergence today (executed): `float32 (1, 1, 4, 5, 6)` vs `float64 (1, 4, 5, 6)` for the same cuboid. This is the behavior-change discussion CONTRIBUTING asks for, and the decision is: **in the coordinated Repair & Deprecation window, not before. A PR that aliases or consolidates these ahead of it will be closed and pointed here.**

Found during the bbox/boxes conventions audit batch 4c (#4020); the audit changes no runtime behavior. If the maintainers prefer, this can be folded into #4015 as a second axis of the same shared-implementation decision.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
